aoc/src/utils.rs

```diff
@@ -33,19 +34,38 @@ where
 
 /// Slurp file will try to parse the string into `T` as long as T implements FromStr
 #[allow(clippy::missing_errors_doc)]
-pub fn slurp_file<P, T>(path: P) -> Result<Vec<T>, SlurpError>
+pub fn slurp_file<P, T>(path: P) -> impl Iterator<Item = Result<T, <T as FromStr>::Err>>
```

Comment:
Converts this to return an iterator now, and use read_until, which should all around be more performant.

It's not going to beat something that is looking for newlines while doing parsing, since this will still look at every byte twice, but it's pretty reasonable.